compiler/grainformat/reformat.re

```diff
@@ -839,30 +878,16 @@ and print_pattern =
     clean_pattern;
   };
 }
-and print_constant = (c: Parsetree.constant) => {
-  let print_c =
-    switch (c) {
-    | PConstNumber(PConstNumberInt(i)) => Printf.sprintf("%s", i)
-    | PConstNumber(PConstNumberFloat(f)) => Printf.sprintf("%s", f)
-    | PConstNumber(PConstNumberRational(n, d)) =>
-      Printf.sprintf("%s/%s", n, d)
-    | PConstBytes(b) => Printf.sprintf("%s", b)
-    | PConstChar(c) => Printf.sprintf("'%s'", String.escaped(c))
-    | PConstFloat64(f) => Printf.sprintf("%sd", f)
-    | PConstFloat32(f) => Printf.sprintf("%sf", f)
-    | PConstInt32(i) => Printf.sprintf("%sl", i)
-    | PConstInt64(i) => Printf.sprintf("%sL", i)
-    | PConstWasmI32(i) => Printf.sprintf("%sn", i)
-    | PConstWasmI64(i) => Printf.sprintf("%sN", i)
-    | PConstWasmF32(f) => Printf.sprintf("%sw", f)
-    | PConstWasmF64(f) => Printf.sprintf("%sW", f)
-    | PConstBool(true) => "true"
-    | PConstBool(false) => "false"
-    | PConstVoid => "void"
-    | PConstString(s) => Printf.sprintf("\"%s\"", String.escaped(s))
-    };
-  Doc.text(print_c);
+
+and print_constant =
+    (
+      ~original_source: array(string),
+      ~loc: Grain_parsing__Location.t,
+      c: Parsetree.constant,
+    ) => {
+  Doc.text(get_original_code_snippet(loc, original_source));
```

Comment:
Might warrant some sort of comment along the lines of "we just grab the original code for constants since it must be well-formatted"

Reply:
Done.